conan/cli/commands/lock.py

```diff
@@ -179,3 +183,101 @@ def lock_update(conan_api, parser, subparser, *args):
+    subparser.add_argument('--update-config-requires', action="append", help='Update config-requires from lockfile')
+    subparser.add_argument('--build-require', action='store_true', default=False, help='Whether the provided reference is a build-require')
+    subparser.add_argument('--transitives', action='store_true', default=False, help='Upgrade also transitive dependencies')
+    args = parser.parse_args(*args)
+
+    # parameter validation
+    validate_common_graph_args(args)
+
+    if not any([args.update_requires, args.update_build_requires, args.update_python_requires, args.update_config_requires]):
+        raise ConanException("At least one of --update-requires, --update-build-requires, "
+                             "--update-python-requires or --update-config-requires should be specified")
+
+    cwd = os.getcwd()
+    path = conan_api.local.get_conanfile_path(args.path, cwd, py=None) if args.path else None
+    remotes = conan_api.remotes.list(args.remote) if not args.no_remote else []
+    overrides = eval(args.lockfile_overrides) if args.lockfile_overrides else None
+    lockfile = conan_api.lockfile.get_lockfile(lockfile=args.lockfile, conanfile_path=path,
+                                               cwd=cwd, partial=True, overrides=overrides)
+    profile_host, profile_build = conan_api.profiles.get_profiles_from_args(args)
+
+    def expand_graph():
+        if path:
+            return conan_api.graph.load_graph_consumer(path, args.name, args.version,
+                                                        args.user, args.channel,
+                                                        profile_host, profile_build, lockfile,
+                                                        remotes, args.update,
+                                                        is_build_require=args.build_require)
+        return conan_api.graph.load_graph_requires(args.requires, args.tool_requires,
+                                                    profile_host, profile_build, lockfile,
+                                                    remotes, args.update)
+    requested_update = {"requires": args.update_requires or [],
+                        "build_requires": args.update_build_requires or [],
+                        "python_requires": args.update_python_requires or [],
+                        "config_requires": args.update_config_requires or []}
+    if args.transitives:
+        def is_version_range(ref: str):
+            # TODO: use conan's built in
+            return ref.find('[') != -1 or ref.find('*') != -1
+
+        def match_version(ref, node):
+            if is_version_range(ref):
+                return VersionRange(str(RecipeReference.loads(ref).version)).contains(Version(node.ref), resolve_prerelease=True)
+            else:
+                return node.ref.matches(ref, is_consumer=None)
+        updatable_deps = defaultdict(list)
+        for node in expand_graph().nodes:
```

Comment:
I think this might be a bit overkill, I'd try to leverage the existing conan ``lock remove`` functionality.